src/main/java/uk/gov/hmcts/reform/cpo/exception/HttpError.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.hmcts.reform.cpo.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.util.UriUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+public class HttpError<T extends Serializable> implements Serializable {
+    public static final Integer DEFAULT_STATUS = HttpStatus.INTERNAL_SERVER_ERROR.value();
+    public static final String DEFAULT_ERROR = "Unexpected Error";
+
+    private final String exception;
+    private final transient LocalDateTime timestamp;
+    private final Integer status;
+    private final String error;
+    private final String message;
+    private final String path;
+    private T details;
+
+    public HttpError(Exception exception, HttpServletRequest request,HttpStatus status) {
+        final ResponseStatus responseStatus = exception.getClass().getAnnotation(ResponseStatus.class);
+
+        this.exception = exception.getClass().getName();
+        this.timestamp = LocalDateTime.now(ZoneOffset.UTC);
+        this.status = getStatusFromResponseStatus(responseStatus,status);
+        this.error = getErrorReason(responseStatus);
+        this.message = exception.getMessage();
+        this.path = UriUtils.encodePath(request.getRequestURI(), StandardCharsets.UTF_8);
```

Comment:
the `HttpServletRequest` is only used to get the path. Probably we could remove this dependency by passing the path directly as a constructor argument, rather than the whole `HttpServletRequest` class. This would make the `HttpError` more reusable and simpler. Also to test

Reply:
TODO